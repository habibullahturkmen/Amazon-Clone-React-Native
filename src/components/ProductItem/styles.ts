import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
    root: {
        flexDirection: 'row',
        borderColor: '#d1d1d1',
        borderWidth: 1,
        borderRadius: 5,
        backgroundColor: '#fff',
        marginVertical: 5,
    },
    image: {
        flex: 2,
        height: 150,
        resizeMode: 'contain',
    },
    rightContainer: {
        flex: 3,
        padding: 10,
    },
    title: {
        fontSize: 18,
    },
    ratingsContainer: {
        flexDirection: 'row',
        alignItems: 'center',
        marginVertical: 5,
    },
    star: {
        margin: 0.5,
    },
    price: {
        fontSize: 18,
        fontWeight: 'bold',
    },
    oldPrice: {
        fontSize: 12,
        fontWeight: 'normal',
        textDecorationLine: 'line-through',
    },
});

export default styles;
