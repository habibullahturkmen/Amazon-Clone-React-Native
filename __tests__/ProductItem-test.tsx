import 'react-native';
import React from 'react';
import ProductItem from '../src/components/ProductItem';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer.create(
        <ProductItem
            item={{
                id: '1',
                title: 'New Book',
                image: '',
                avgRating: 10,
                ratings: 20,
                price: 100,
                oldPrice: 99,
            }}
        />,
    );
    expect(tree).toMatchSnapshot();
});

it('has 1 child', () => {
    const tree = renderer
        .create(
            <ProductItem
                item={{
                    id: '1',
                    title: 'New Book',
                    image: '',
                    avgRating: 10,
                    ratings: 20,
                    price: 100,
                    oldPrice: 99,
                }}
            />,
        )
        .toJSON();
    // @ts-ignore: Object is possibly 'null'
    expect(tree.children.length).toBe(2);
});
