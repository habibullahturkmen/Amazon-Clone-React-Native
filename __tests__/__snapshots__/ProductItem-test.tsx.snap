// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "#d1d1d1",
      "borderRadius": 5,
      "borderWidth": 1,
      "flexDirection": "row",
      "marginVertical": 5,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "",
      }
    }
    style={
      Object {
        "flex": 2,
        "height": 150,
        "resizeMode": "contain",
      }
    }
  />
  <View
    style={
      Object {
        "flex": 3,
        "padding": 10,
      }
    }
  >
    <Text
      numberOfLines={3}
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      New Book
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginVertical": 5,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#e47911",
              "fontSize": 18,
            },
            Object {
              "margin": 0.5,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#e47911",
              "fontSize": 18,
            },
            Object {
              "margin": 0.5,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#e47911",
              "fontSize": 18,
            },
            Object {
              "margin": 0.5,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#e47911",
              "fontSize": 18,
            },
            Object {
              "margin": 0.5,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#e47911",
              "fontSize": 18,
            },
            Object {
              "margin": 0.5,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text>
         
        20
      </Text>
    </View>
    <Text
      style={
        Object {
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      from $
      100
      <Text
        style={
          Object {
            "fontSize": 12,
            "fontWeight": "normal",
            "textDecorationLine": "line-through",
          }
        }
      >
         $
        99
         
      </Text>
    </Text>
  </View>
</View>
`;
